src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -180,29 +166,34 @@ internal bool TryReinitializeParser()
 
             StopParser();
 
-            if (!_documentTracker.IsSupportedProject)
-            {
-                // Tracker is either starting up, tearing down or wrongfully instantiated.
-                // Either way, the tracker can't act on its associated project, neither can we.
-                return false;
-            }
-
             StartParser();
 
             return true;
         }
 
-        // Internal for testing
-        internal void StartParser()
+        private void StartParser()
         {
             _dispatcher.AssertForegroundThread();
 
-            // Make sure any tests use the real thing or a good mock. These tests can cause failures
-            // that are hard to understand when this throws.
-            Debug.Assert(_documentTracker.IsSupportedProject);
-            Debug.Assert(_documentTracker.ProjectSnapshot != null);
+            if (_parser != null)
+            {
+                Debug.Fail("Parser has already been started.");
+                return;
+            }
 
-            _projectEngine = _projectEngineFactory.Create(_documentTracker.ProjectSnapshot, ConfigureProjectEngine);
+            if (_documentTracker.IsSupportedProject)
+            {            
+                // Make sure any tests use the real thing or a good mock. These tests can cause failures
+                // that are hard to understand when this throws.
+                Debug.Assert(_documentTracker.ProjectSnapshot != null);
+                Debug.Assert(_documentTracker.Workspace != null);
+                var projectEngineFactory = _documentTracker.Workspace.Services.GetRequiredService<ProjectSnapshotProjectEngineFactory>();
+                _projectEngine = projectEngineFactory.Create(_documentTracker.ProjectSnapshot, ConfigureProjectEngine);
+            }
+            else
+            {
+                _projectEngine = RazorProjectEngine.Create((builder) => { });
```

Comment:
The initial startup experience (until a workspace is available from a C# buffer) is a bare minimum parser that doesn't have directives such as `@model`, doesn't have TagHelpers etc.